subsys/caf/modules/buttons.c

```diff
@@ -51,8 +59,20 @@ static enum state state;
 
 static void scan_fn(struct k_work *work);
 
+static uint32_t mask_wakeup(const struct gpio_pin *pins, size_t cnt)
+{
+	uint32_t mask = MASK_ALL;
 
-static int set_cols(uint32_t mask)
+	for (size_t i = 0; i < cnt; i++) {
+		if (pins[i].wakeup_blocked) {
+			mask &= ~BIT(i);
+		}
+	}
+
+	return mask;
+}
+
+static int set_cols(uint32_t mask, bool enforce_state)
```

Comment:
Maybe we can call it `enforce_pin_state` (so that pins are guaranteed not to be floating)? Otherwise the option would be silently ignored if pin is set as an output and API usage may be slightly counterintuitive.

Reply:
don't know. the original name was not intuitive. I think it's best to call it what it is - pull control. Note that some people may not want pull at all.